Bootstrap/app/src/main/java/ar/com/wolox/android/example/ui/logIn/LogInFragment.java

```diff
@@ -0,0 +1,80 @@
+package ar.com.wolox.android.example.ui.logIn;
+
+
+import android.content.Intent;
+import android.text.TextUtils;
+import android.widget.Button;
+import android.widget.EditText;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+import ar.com.wolox.android.R;
+import ar.com.wolox.android.example.ui.home.HomeActivity;
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment;
+import butterknife.BindView;
+import butterknife.OnClick;
+
+public class LogInFragment extends WolmoFragment<LogInPresenter> implements ILogInView{
+
+    @Inject
+    public LogInFragment(){
+    }
+    @BindView(R.id.emailText)
+    EditText emailText;
+    @BindView(R.id.passwordText) EditText passwordText;
+    @BindView (R.id.logIn)
+    Button logInButton;
+    @BindView (R.id.signUp) Button signUpButton;
+
+
+
+    @Override
+    public int layout() {
+        return R.layout.fragment_login;
+    }
+
+    @Override
+    public void init() {
+    }
+
+    private boolean validarEmail(String email){
+        return  !TextUtils.isEmpty(email) && android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches();
+    }
+    private boolean validarPassword (String password){
+        Pattern pattern;
+        Matcher matcher;
+        final String PASSWORD_PATTERN = "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9]).{8,}$";
```

Comment:
checking the password it's not required. But next time you can create a new Class like PatternMatcherHelper or something like that , that does the work.